### PR TITLE
Code Cleanup

### LINKED ISSUE #200: [Search implementation] making search queries addressable via URLs
Requirement: 
To make search links for keywords shareable

Approach: Leveraging the `history.pushState` API to update the URL (add as query parameter) without reloading the page.
https://main--fonterra-fernleaf--hlxsites.hlx.live/en?search=foo

### PR DESCRIPTION
**Pull request by noormd08**
Please always provide the [GitHub issue(s)](../issues) your PR is for, as well as test URLs where your change can be observed (before and after):

Fix #200 

Test URLs:
- Before: https://main--fonterra-fernleaf--hlxsites.hlx.page/en
- After: https://200-search-query--fonterra-fernleaf--hlxsites.hlx.page/en

### CODE REVIEW

File: blocks/header/search.js [hunk 1]
```diff
@@ -109,7 +106,7 @@ export default function Search() {
       ? `<h6><a href='${item.path}'>${item.title}</a></h6>`
       : `<span class='item-img'><a href='${item.path}'>
           <picture>
-              <img src='${removeImageProps(item.image)}'>
+              <img src='${removeImageProps(item.image)}' alt='${item.title}' />
```
Comment (sselvara) [inline blocks/header/search.js:109]: use a relevant function name and modify the logic used in the function with URL object logic.

File: blocks/header/search.js [hunk 2]
```diff
@@ -176,10 +173,9 @@ export default function Search() {
       history.pushState(null, null, currentUrl.toString());
       const spinner = this.dialogElem.querySelector('.overlay-loading');
       spinner.style.display = 'block';
-      const placeholders = await fetchPlaceholders();
       const searchData = await fetchSearch();
       spinner.style.display = 'none';
-      if (placeholders && searchData) {
+      if (searchData && searchData.length) {
```
Comment (sselvara) [inline blocks/header/search.js:178]: simplify this with Optional chaining (?.) operator, if(searchData?.length)

**Review by sselvara — CHANGES_REQUESTED** [review]

File: blocks/header/search.js (hunk 2 above)
Comment (noormd08) [reply blocks/header/search.js]: Fixed

File: blocks/header/search.js (hunk 1 above)
Comment (noormd08) [reply blocks/header/search.js:109]: As discussed, no changes required.

**Comment by asthabh23** [comment]
@noormd08 best practices score is reduced, please see
https://pagespeed.web.dev/analysis/https-200-search-query--fonterra-fernleaf--hlxsites-hlx-live-en/fay5hwtd7t?form_factor=mobile

**Review by sselvara — APPROVED** [review]

**Comment by asthabh23** [comment]
@sselvara , please have a look at my comment above before merging

**Comment by sselvara** [comment]
@asthabh23  please check https://pagespeed.web.dev/analysis/https-200-search-query--fonterra-fernleaf--hlxsites-hlx-live-en/6e7uvkh506?form_factor=mobile

**Comment by sselvara** [comment]
@asthabh23  Best practices score is 100 for me most of the time, please check https://pagespeed.web.dev/analysis/https-200-search-query--fonterra-fernleaf--hlxsites-hlx-live-en/cmlnmgzx5l?form_factor=mobile

**Pull request merged**